### ISSUE TITLE
redundantReturn removes return statement when it shouldn't

### ISSUE THREAD

**Issue by dirkoswanepoel** [nicklockwood/SwiftFormat#556]
The first block of code below will not be altered by SwiftFormat. The second block of code is definitely not the cleanest way to write the specific code, but it does compile and run without any issues. Running SwiftFormat however removes the return statement, resulting in execution continuing after the if statement even when the if statement was true. I believe the body of the if statement is treated as a closure.

The code below won't get changed by SwiftFormat:

```
let arr: [CLLocationCoordinate2D] = [
    CLLocationCoordinate2D(latitude: 10.0, longitude: 10.0),
    CLLocationCoordinate2D(latitude: 15.0, longitude: 15.0),
    CLLocationCoordinate2D(latitude: 20.0, longitude: 20.0)
]

if !arr.contains(where: { $0.latitude == 10.0 && $0.longitude == 10.0 }) {
    return true
}
```

The following code will have `return true` replace by `true`:

```
let arr: [CLLocationCoordinate2D] = [
    CLLocationCoordinate2D(latitude: 10.0, longitude: 10.0),
    CLLocationCoordinate2D(latitude: 15.0, longitude: 15.0),
    CLLocationCoordinate2D(latitude: 20.0, longitude: 20.0)
]

if !arr.contains { $0.latitude == 10.0 && $0.longitude == 10.0 } {
    return true
}
```

**Comment by nicklockwood** [comment]
@dirkoswanepoel thanks for reporting this, I'll try to fix it asap.

**Comment by nicklockwood** [comment]
@dirkoswanepoel fixed in 0.44.4
